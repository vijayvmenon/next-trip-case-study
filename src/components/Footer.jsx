import React from 'react';
import { createUseStyles } from 'react-jss';
import constants, { colors } from '../constants';

const useStyles = createUseStyles({
  footerRoot: {
    background: colors.BACKGROUND_1,
    height: 75,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  footerText: {
    fontSize: '0.875rem',
    color: colors.GREY,
    lineHeight: '20px',
    textAlign: 'center',
  },
});
export default function Footer() {
  const classes = useStyles();
  return (
    <footer className={classes.footerRoot}>
      <div className={classes.footerText}>
        {constants.FOOTER_TEXT_1}
        <br />
        {constants.FOOTER_TEXT_2}
        <br />
        {constants.FOOTER_TEXT_3}
      </div>
    </footer>
  );
}
