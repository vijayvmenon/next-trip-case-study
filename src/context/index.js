import { useContext } from 'react';
import AppContext, { Context } from './context';

export default AppContext;
export { Context };
export const useMyContext = () => useContext(Context);
