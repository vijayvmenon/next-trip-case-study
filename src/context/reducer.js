import constants from '../constants';
import ENTranslations from '../locales/en.json';
import { localStorageExists } from '../utils';

const localStorageLanguage =
  localStorageExists && localStorage.getItem('LANGUAGE');
const initialState = {
  tableExpanded: false,
  hideHeader: false,
  snackbar: { value: '', text: '' },
  language: localStorageLanguage || 'en',
  translations: localStorageLanguage
    ? constants.LANGUAGE_MAPPING[localStorageLanguage]
    : ENTranslations,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'TABLE_EXPAND':
      return { ...state, tableExpanded: true };
    case 'TABLE_COLLAPSE':
      return { ...state, tableExpanded: false };
    case 'SET_SNACKBAR':
      return { ...state, snackbar: { value: action.value, text: action.text } };
    case 'HIDE_HEADER':
      return { ...state, hideHeader: true };
    case 'SHOW_HEADER':
      return { ...state, hideHeader: false };
    case 'SET_LANGUAGE':
      return {
        ...state,
        language: action.value,
        translations: constants.LANGUAGE_MAPPING[action.value],
      };
    default:
      return { ...state };
  }
}

export { initialState };
