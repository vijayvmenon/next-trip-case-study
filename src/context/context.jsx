import React, { useReducer, createContext } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';

export const Context = createContext();

export default function AppContext({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
}

AppContext.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,
};
