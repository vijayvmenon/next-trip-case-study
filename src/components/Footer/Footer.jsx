import React, { useContext } from 'react';
import { createUseStyles } from 'react-jss';
import constants, { colors } from '../../constants';
import { Context } from '../../context';

const useStyles = createUseStyles({
  footerRoot: {
    background: colors.BACKGROUND_1,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: '0.875rem 1.6rem',
  },
  footerText: {
    fontSize: '0.875rem',
    color: colors.GREY,
    lineHeight: '20px',
    // textAlign: 'center',
    '@media(max-width:600px)': {
      fontSize: '0.7rem',
    },
    marginRight: '2rem',
  },
  copyRightText: {
    fontSize: '0.875rem',
    color: colors.GREY,
    lineHeight: '20px',
    textAlign: 'center',
    '@media(max-width:600px)': {
      fontSize: '0.7rem',
    },
  },
});
export default function Footer() {
  const classes = useStyles();
  const { state: { translations = {} } = {} } = useContext(Context);

  return (
    <footer className={classes.footerRoot}>
      <div className={classes.footerText}>
        {translations.FOOTER_TEXT_1}
        {translations.FOOTER_TEXT_2}
      </div>
      <div className={classes.copyRightText}>{constants.FOOTER_TEXT_3}</div>
    </footer>
  );
}
