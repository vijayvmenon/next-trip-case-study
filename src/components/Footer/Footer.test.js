import Footer from './Footer';
import renderer from 'react-test-renderer';
import { Context } from '../../context';

const dispatch = jest.fn(); // mock dispatch

describe('Component:Footer.jsx', () => {
  test('Footer.jsx renders correctly', () => {
    const footerTree = renderer
      .create(
        <Context.Provider value={{ dispatch }}>
          <Footer />
        </Context.Provider>,
      )
      .toJSON();
    expect(footerTree).toMatchSnapshot();
  });
});
