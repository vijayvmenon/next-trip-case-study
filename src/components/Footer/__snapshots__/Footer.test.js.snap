// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Footer.jsx Footer.jsx renders correctly 1`] = `
<footer
  className="footerRoot-0-2-1"
>
  <div
    className="footerText-0-2-2"
  />
  <div
    className="copyRightText-0-2-3"
  >
    Â© 2021 Metro Transit
  </div>
</footer>
`;
